refs/heads/master: kv/kvserver/asim/tests: TestDataDriven failed

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/21191100?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/21191100?buildTab=artifacts#/) on refs/heads/master @ [d3b44e6555155fc65d0dd6776b58ac16f61fe391](https://github.com/cockroachdb/cockroach/commits/d3b44e6555155fc65d0dd6776b58ac16f61fe391):


```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2616/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt:23:
          set_capacity
          ----
    datadriven_simulation_test.go:618: running eval for fulldisk
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2616/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt:29: still running after 10.000421932s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2616/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt:29: still running after 20.000771822s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2616/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt:29: still running after 30.001771934s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2616/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt:29: still running after 40.002582953s
    datadriven_simulation_test.go:233: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2616/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt:29:
          eval
        output didn't match expected:
          @@ -1,28 +1,28 @@
           sma-count:
          -disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.05] (stddev=0.34, mean=0.37, sum=2)
          +disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.00] (stddev=0.32, mean=0.36, sum=2)
           disk_fraction_used#1: last:  [s1=0.27, s2=0.27, s3=0.27, s4=0.27, s5=0.94] (stddev=0.27, mean=0.40, sum=2)
          -disk_fraction_used#1: thrash_pct: [s1=19%, s2=20%, s3=20%, s4=22%, s5=67%]  (sum=149%)
          +disk_fraction_used#1: thrash_pct: [s1=20%, s2=22%, s3=21%, s4=23%, s5=68%]  (sum=155%)
           replicas#1: first: [s1=300, s2=300, s3=300, s4=300, s5=300] (stddev=0.00, mean=300.00, sum=1500)
           replicas#1: last:  [s1=319, s2=317, s3=324, s4=323, s5=217] (stddev=41.58, mean=300.00, sum=1500)
           replicas#1: thrash_pct: [s1=185%, s2=197%, s3=197%, s4=213%, s5=43%]  (sum=835%)
          -hash: 8473bd9f50baf595
          +hash: c3f26592a16d40d7
           ==========================
           mma-only:
          -disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.05] (stddev=0.34, mean=0.37, sum=2)
          +disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.00] (stddev=0.32, mean=0.36, sum=2)
           disk_fraction_used#1: last:  [s1=0.28, s2=0.28, s3=0.28, s4=0.26, s5=0.90] (stddev=0.25, mean=0.40, sum=2)
           disk_fraction_used#1: thrash_pct: [s1=1%, s2=0%, s3=1%, s4=0%, s5=47%]  (sum=48%)
           replicas#1: first: [s1=300, s2=300, s3=300, s4=300, s5=300] (stddev=0.00, mean=300.00, sum=1500)
           replicas#1: last:  [s1=329, s2=330, s3=329, s4=305, s5=207] (stddev=47.45, mean=300.00, sum=1500)
           replicas#1: thrash_pct: [s1=4%, s2=0%, s3=6%, s4=0%, s5=0%]  (sum=11%)
          -hash: 491e5ffd11727f20
          +hash: 7b3a6b16bb384000
           ==========================
           mma-count:
          -disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.05] (stddev=0.34, mean=0.37, sum=2)
          +disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.00] (stddev=0.32, mean=0.36, sum=2)
           disk_fraction_used#1: last:  [s1=0.28, s2=0.28, s3=0.27, s4=0.27, s5=0.83] (stddev=0.22, mean=0.39, sum=2)
          -disk_fraction_used#1: thrash_pct: [s1=23%, s2=21%, s3=23%, s4=27%, s5=46%]  (sum=141%)
          +disk_fraction_used#1: thrash_pct: [s1=24%, s2=22%, s3=25%, s4=29%, s5=46%]  (sum=147%)
           replicas#1: first: [s1=300, s2=300, s3=300, s4=300, s5=300] (stddev=0.00, mean=300.00, sum=1500)
           replicas#1: last:  [s1=329, s2=329, s3=325, s4=324, s5=193] (stddev=53.54, mean=300.00, sum=1500)
           replicas#1: thrash_pct: [s1=179%, s2=166%, s3=183%, s4=215%, s5=0%]  (sum=743%)
          -hash: 1f619c1de43e855f
          +hash: d5529a0f915662df
           ==========================
           
--- FAIL: TestDataDriven/mode=sma/fulldisk (41.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61018

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/sma/fulldisk.txt
@@ -29,29 +29,29 @@ set_capacity store=5 capacity=107374182400
 eval duration=20m seed=42 metrics=(replicas,disk_fraction_used) cfgs=(sma-count,mma-only,mma-count)
 ----
 sma-count:
-disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.05] (stddev=0.34, mean=0.37, sum=2)
+disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.00] (stddev=0.32, mean=0.36, sum=2)
 disk_fraction_used#1: last:  [s1=0.27, s2=0.27, s3=0.27, s4=0.27, s5=0.94] (stddev=0.27, mean=0.40, sum=2)
-disk_fraction_used#1: thrash_pct: [s1=19%, s2=20%, s3=20%, s4=22%, s5=67%]  (sum=149%)
+disk_fraction_used#1: thrash_pct: [s1=20%, s2=22%, s3=21%, s4=23%, s5=68%]  (sum=155%)
 replicas#1: first: [s1=300, s2=300, s3=300, s4=300, s5=300] (stddev=0.00, mean=300.00, sum=1500)
 replicas#1: last:  [s1=319, s2=317, s3=324, s4=323, s5=217] (stddev=41.58, mean=300.00, sum=1500)
 replicas#1: thrash_pct: [s1=185%, s2=197%, s3=197%, s4=213%, s5=43%]  (sum=835%)
-hash: 8473bd9f50baf595
+hash: c3f26592a16d40d7
 ==========================
 mma-only:
-disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.05] (stddev=0.34, mean=0.37, sum=2)
+disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.00] (stddev=0.32, mean=0.36, sum=2)
 disk_fraction_used#1: last:  [s1=0.28, s2=0.28, s3=0.28, s4=0.26, s5=0.90] (stddev=0.25, mean=0.40, sum=2)
 disk_fraction_used#1: thrash_pct: [s1=1%, s2=0%, s3=1%, s4=0%, s5=47%]  (sum=48%)
 replicas#1: first: [s1=300, s2=300, s3=300, s4=300, s5=300] (stddev=0.00, mean=300.00, sum=1500)
 replicas#1: last:  [s1=329, s2=330, s3=329, s4=305, s5=207] (stddev=47.45, mean=300.00, sum=1500)
 replicas#1: thrash_pct: [s1=4%, s2=0%, s3=6%, s4=0%, s5=0%]  (sum=11%)
-hash: 491e5ffd11727f20
+hash: 7b3a6b16bb384000
 ==========================
 mma-count:
-disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.05] (stddev=0.34, mean=0.37, sum=2)
+disk_fraction_used#1: first: [s1=0.20, s2=0.20, s3=0.20, s4=0.20, s5=1.00] (stddev=0.32, mean=0.36, sum=2)
 disk_fraction_used#1: last:  [s1=0.28, s2=0.28, s3=0.27, s4=0.27, s5=0.83] (stddev=0.22, mean=0.39, sum=2)
-disk_fraction_used#1: thrash_pct: [s1=23%, s2=21%, s3=23%, s4=27%, s5=46%]  (sum=141%)
+disk_fraction_used#1: thrash_pct: [s1=24%, s2=22%, s3=25%, s4=29%, s5=46%]  (sum=147%)
 replicas#1: first: [s1=300, s2=300, s3=300, s4=300, s5=300] (stddev=0.00, mean=300.00, sum=1500)
 replicas#1: last:  [s1=329, s2=329, s3=325, s4=324, s5=193] (stddev=53.54, mean=300.00, sum=1500)
 replicas#1: thrash_pct: [s1=179%, s2=166%, s3=183%, s4=215%, s5=0%]  (sum=743%)
-hash: 1f619c1de43e855f
+hash: d5529a0f915662df
 ==========================

